frame/system/src/extras.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+//! Strongly typed interface for SignedExtensions
+//!
+
+use sp_runtime::traits::{StaticLookup, SignedExtension};
+use sp_runtime::generic::Era;
+
+/// AccountIndex type for Runtime
+pub type IndexFor<R> = <R as crate::Trait>::Index;
+/// Call type for Runtime
+pub type CallFor<R> = <R as crate::Trait>::Call;
+/// Address type for runtime.
+pub type AddressFor<R> = <<R as crate::Trait>::Lookup as StaticLookup>::Source;
+/// Hash for runtime.
+pub type HashFor<R> = <R as crate::Trait>::Hash;
+/// AccountId type for runtime.
+pub type AccountIdFor<R> = <R as crate::Trait>::AccountId;
+
+///
+/// Strongly typed interface for the `SignedExtensions` that should be included in extrinsics
+/// for them to valid for your runtime.
+///
+/// ```rust,ignore
```

Comment:
Make it compile or it is almost useless, because it gets outdated really fast

Reply:
> Make it compile

unfortunately, this is impossible as that requires a full runtime so instead, I removed it.